### ISSUE TITLE
EWS.HTTP.Get_Line and buffer boundary

### ISSUE THREAD

**Issue by jquorning** [simonjwright/ews#1]
This issue is tested with the 2019 archive from SourceForge.

It looks like `Get_Line` is misbehaving when the `Line` buffer is smaller than the input. At least it behaves different than `Ada.Text_IO.Get_Line`.

I have changed the `Upload` procedure in `AJAX.adb` to basically contain:
```Ada
   Line : String (1 .. 4);
   Last : Natural;
begin
   EWS.HTTP.Open (Curs, Attachments);
   while not EWS.HTTP.End_Of_File (Curs) loop
      EWS.HTTP.Get_Line (Curs, Line, Last);
      Put_Line (Line (Line'First .. Last));
   end loop;
```
and upload
```
1
12
123
1234
12345
123456
1234567
12345678
123456789
1234567890
```
The output is:
```
1
12
123
1234
1234

1234
6
1234
67
1234
678
1234
6789
1234
6789

```
The `5`s and the `0`s are missing at Line boundary.
The `Ada.Text_IO.Get_Line` behaves like:
```
1
12
123
1234

1234
5
1234
56
1234
567
1234
5678

1234
5678
9
1234
5678
90
```
With this program:
```Ada
with Ada.Text_IO; use Ada.Text_IO;
procedure Stand is
   Line : String (1 .. 4);
   Last : Natural;
begin
   while not End_Of_File loop
      Get_Line (Line, Last);
      Put_Line (Line (Line'First .. Last));
   end loop;
end Stand;
```

**Comment by simonjwright** [comment]
Not quite sure what your use case was?

I certainly think that I gave up too soon with attachments. What if they contain binary?

At the very least, I could check whether the `Cursor`’s contents appear to contain text (e.g. a line terminator within the first 256 characters?) before allowing `Get_Line`.

**Comment by simonjwright** [comment]
Thanks for the report; took me a while to see the issue, sorry.

Fixed in 197da73.
